File: server/Services/IngredientsService.cs
namespace all_state.Services;


public class IngredientsService
{
  private readonly IngredientsRepository _ingredientsRepository;
  private readonly RecipesService _recipesService;


  public IngredientsService(IngredientsRepository ingredientsRepository, RecipesService recipesService)
  {
    _ingredientsRepository = ingredientsRepository;
    _recipesService = recipesService;
  }

  public Ingredient CreateIngredient(Ingredient ingredientData, string userId)
  {
    Recipe recipe = _recipesService.GetRecipeById(ingredientData.RecipeId);
    if (recipe.CreatorId != userId)
    {
      throw new Exception("No add ingredients to the recipes of others.");
    }
    Ingredient ingredient = _ingredientsRepository.CreateIngredient(ingredientData);
    return ingredient;
  }
}

File: client/src/services/IngredientsService.js
import { AppState } from "@/AppState.js"
import { api } from "./AxiosService.js"
import { Ingredient } from "@/models/Ingredient.js"



class IngredientsService{


  async getIngredientsByRecipe(recipeId) {
    AppState.ingredients = []
    const response = await api.get(`api/recipes/${recipeId}/ingredients`)
    AppState.ingredients = response.data.map((ingredient) => new Ingredient(ingredient))
  }



}





export const ingredientsService = new IngredientsService()





